src/components/related_items_outfit/YourOutfit.jsx

```diff
@@ -1,7 +1,21 @@
-import React from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
 import AddOutfitCard from './AddOutfitCard';
+import Carousel from './Carousel';
+import ProductCard from './RIProductCard';
+import Compare from './CompareModal';
+
+export default function YourOutfit({ curProd }) {
+  const [outfit, setOutfit] = useState([]);
+  const [openModal, setOpenModal] = useState(false);
+
+  const handleAdd = () => {
+    const newOutfit = outfit;
+    newOutfit.push(curProd);
+    setOutfit(newOutfit);
+    console.log(outfit);
```

Comment:
You should avoid directly using state to create a new state, instead consider using the callback functionality for the setter function:
`setOutfit(prevOutfit => [...prevOutfit, currProd])`

If you want to include the console log for testing, you can expand it:
`setOutfit(prevOutfit => {
  console.log([...prevOutfit, currProd]);
  return [...prevOutfit, currProd];
})`

Reply:
Updated, and pushing the other work from this morning with it.